core/capabilities/syncer.go

```diff
@@ -56,26 +71,486 @@ var defaultStreamConfig = p2ptypes.StreamConfig{
+					return nil, err
+				}
+
+				aggregator := triggers.NewMercuryRemoteAggregator(
+					codec,
+					signers,
+					int(remoteDON.F+1),
+					s.lggr,
+				)
+				cfg := &remotetypes.RemoteTriggerConfig{}
+				cfg.ApplyDefaults()
+				err = proto.Unmarshal(c.Config, cfg)
+				if err != nil {
+					return nil, err
+				}
+				// TODO: We need to implement a custom, Mercury-specific
+				// aggregator here, because there is no guarantee that
+				// all trigger events in the workflow will have the same
+				// payloads. As a workaround, we validate the signatures.
+				// When this is solved, we can move to a generic aggregator
+				// and remove this.
+				triggerCap := remote.NewTriggerSubscriber(
+					cfg,
+					info,
+					*toDONInfo(remoteDON),
+					*toDONInfo(myDON),
+					s.dispatcher,
+					aggregator,
+					s.lggr,
+				)
+				return triggerCap, nil
+			}
+			err := s.addToRegistryAndSetDispatcher(ctx, capability, remoteDON, newTriggerFn)
+			if err != nil {
+				return fmt.Errorf("failed to add trigger shim: %w", err)
+			}
+		case capabilities.CapabilityTypeAction:
+			s.lggr.Warn("no remote client configured for capability type action, skipping configuration")
+		case capabilities.CapabilityTypeConsensus:
+			s.lggr.Warn("no remote client configured for capability type consensus, skipping configuration")
+		case capabilities.CapabilityTypeTarget:
+			newTargetFn := func(info capabilities.CapabilityInfo) (capabilityService, error) {
+				client := target.NewClient(
+					info,
+					*toDONInfo(remoteDON),
+					s.dispatcher,
+					defaultTargetRequestTimeout,
+					s.lggr,
+				)
+				return client, nil
+			}
+
+			err := s.addToRegistryAndSetDispatcher(ctx, capability, remoteDON, newTargetFn)
+			if err != nil {
+				return fmt.Errorf("failed to add target shim: %w", err)
+			}
+		default:
+			s.lggr.Warnf("unknown capability type, skipping configuration: %+v", capability)
+		}
+	}
+	return nil
+}
+
+type capabilityService interface {
+	capabilities.BaseCapability
+	remotetypes.Receiver
+	services.Service
+}
+
+func (s *registrySyncer) addToRegistryAndSetDispatcher(ctx context.Context, capabilityInfo kcr.CapabilityRegistryCapability, don kcr.CapabilityRegistryDONInfo, newCapFn func(info capabilities.CapabilityInfo) (capabilityService, error)) error {
+	fullCapID := fmt.Sprintf("%s@%s", capabilityInfo.LabelledName, capabilityInfo.Version)
+	info, err := capabilities.NewRemoteCapabilityInfo(
+		fullCapID,
+		toCapabilityType(capabilityInfo.CapabilityType),
+		fmt.Sprintf("Remote Capability for %s", fullCapID),
+		toDONInfo(don),
+	)
+	if err != nil {
+		return err
+	}
+	capability, err := newCapFn(info)
+	if err != nil {
+		return err
+	}
+
+	err = s.registry.Add(ctx, capability)
+	if err != nil {
+		// If the capability already exists, we've handled this in
+		// a previous syncer iteration, let's skip and move on
+		// to other capabilities.
+		if errors.Is(err, ErrCapabilityAlreadyExists) {
+			return nil
+		}
+
+		return err
+	}
```

Comment:
Could we have an `s.registry.DoesCapabilityExist` check instead? I suspect we'll need it in many other places, too, e.g., when we want to validate that we have all the capabilities necessary to support a workflow spec.

Reply:
We do have that, you can just call `registry.Get`

Reply:
`Get` returns a capability. Checking on the error type for existence seems wrong. This is just my opinion, though, so 🤷‍♂️